paymentsheet-example/src/androidTest/java/com/stripe/android/lpm/TestCard.kt

```diff
@@ -60,4 +66,86 @@ internal class TestCard : BasePlaygroundTest() {
             )
         )
     }
+
+    /*
+     * TODO(samer-stripe): Once we update `PaymentResult` to return a `StripeIntent`, update the test
+     *  to check against the payment method IDs rather than the last 4 digits.
+     */
+    @OptIn(ExperimentalTestApi::class)
+    @Test
+    fun testDefaultSavedPaymentMethodUsedAfterSingleSave() {
+        val cardNumber = "6011111111111117"
+        val testParameters = TestParameters.create(
+            paymentMethod = LpmRepository.HardcodedCard,
+        ).copy(
+            authorizationAction = null,
+            saveForFutureUseCheckboxVisible = true,
+            saveCheckboxValue = true,
+        )
+
+        val state = testDriver.confirmNewOrGuestComplete(
+            testParameters = testParameters,
+            values = FieldPopulator.Values(
+                cardNumber = cardNumber,
+            ),
+        )
+
+        testDriver.confirmCompleteWithDefaultSavedPaymentMethod(
+            customerId = state?.customerConfig?.id,
+            testParameters = testParameters,
+            beforeBuyAction = { selectors ->
+                selectors.composeTestRule.waitUntilAtLeastOneExists(
+                    hasTestTag(SAVED_PAYMENT_OPTION_TEST_TAG)
+                        .and(isSelected())
+                        .and(hasText(cardNumber.takeLast(4), substring = true))
+                )
+            },
+        )
+    }
+
+    /*
+     * TODO(samer-stripe): Once we update `PaymentResult` to return a `StripeIntent`, update the test
+     *  to check against the payment method IDs rather than the last 4 digits.
+     */
+    @OptIn(ExperimentalTestApi::class)
+    @Test
+    fun testDefaultSavedPaymentMethodUsedAfterMultipleSaves() {
+        val firstCardNumber = "6011111111111117"
+        val secondCardNumber = "6011000990139424"
+
+        val testParameters = TestParameters.create(
+            paymentMethod = LpmRepository.HardcodedCard,
+        ).copy(
+            authorizationAction = null,
+            saveForFutureUseCheckboxVisible = true,
+            saveCheckboxValue = true,
+        )
+
+        val state = testDriver.confirmNewOrGuestComplete(
+            testParameters = testParameters,
+            values = FieldPopulator.Values(
+                cardNumber = firstCardNumber,
+            ),
+        )
+
+        testDriver.confirmExistingComplete(
+            customerId = state?.customerConfig?.id,
+            testParameters = testParameters,
+            values = FieldPopulator.Values(
+                cardNumber = secondCardNumber,
+            ),
+        )
+
+        testDriver.confirmCompleteWithDefaultSavedPaymentMethod(
+            customerId = state?.customerConfig?.id,
+            testParameters = testParameters,
+            beforeBuyAction = { selectors ->
+                selectors.composeTestRule.waitUntilAtLeastOneExists(
```

Comment:
We wouldn't want two of these (that would be a bug). Can we update this to be one and only one?